File: src/main/java/com/example/climbingapi/mapper/WallMapper.java
package com.example.climbingapi.mapper;

import com.example.climbingapi.dto.WallResponse;
import com.example.climbingapi.model.Wall;
import org.springframework.stereotype.Component;

@Component
public class WallMapper {

    public WallResponse toResponse(Wall wall) {
        return new WallResponse(
                wall.getId(),
                wall.getAreaId(),
                wall.getName(),
                wall.getDescription(),
                wall.getLatitude(),
                wall.getLongitude(),
                wall.getApproachInfo(),
                wall.getCreatedAt()
        );
    }
}

File: src/main/kotlin/com/example/climbingapi/mapper/WallMapper.kt
package com.example.climbingapi.mapper

import com.example.climbingapi.dto.WallResponse
import com.example.climbingapi.model.Wall
import com.example.climbingapi.service.StorageService
import org.springframework.stereotype.Component

@Component
class WallMapper(
    private val storageService: StorageService
) {

    fun toResponse(wall: Wall): WallResponse {
        return WallResponse(
            id = wall.id!!,
            areaId = wall.areaId!!,
            name = wall.name!!,
            description = wall.description,
            latitude = wall.latitude,
            longitude = wall.longitude,
            approachInfo = wall.approachInfo,
            imageUrl = (wall.optimizedKey ?: wall.imageKey)?.let { storageService.presignGet(it) },
            thumbnailUrl = (wall.thumbnailKey ?: wall.imageKey)?.let { storageService.presignGet(it) },
            createdAt = wall.createdAt!!
        )
    }
}
